roomserver/storage/shared/storage.go

```diff
@@ -674,6 +674,25 @@ func (d *Database) GetPublishedRooms(ctx context.Context) ([]string, error) {
 	return d.PublishedTable.SelectAllPublishedRooms(ctx, nil, true)
 }
 
+func (d *Database) MissingAuthPrevEvents(
+	ctx context.Context, e *gomatrixserverlib.Event,
+) (missingAuth, missingPrev []string, err error) {
+	for _, authEventID := range e.AuthEventIDs() {
+		if nids, err := d.EventNIDs(ctx, []string{authEventID}); err != nil || len(nids) == 0 {
```

Comment:
For `EventNIDs` we probably can, since that should just omit anything missing, but we can't for `StateAtEventIDs` since that errors if anything is missing.